metabulo/plot.py

```diff
@@ -17,3 +18,17 @@ def make_box_plot(frame):
     output = io.BytesIO()
     FigureCanvas(fig1).print_png(output)
     return output.getvalue()
+
+
+def pca(metabolites, metadata, x_component_index, y_component_index):
+    n_components = max(x_component_index, y_component_index) + 1
+    components = PCA(n_components=n_components).fit_transform(metabolites)
+    samples = []
+    for i in range(metabolites.shape[0]):
+        samples.append({
+            'x': components[i][x_component_index],
+            'y': components[i][y_component_index],
+            'key': metabolites.index[i],
+            'labels': metadata.iloc[i].to_dict()
+        })
+    return samples
```

Comment:
What's your gut on writing a Marshmallow schema here to load this into?  I thought about it, but it seemed like overkill.

Reply:
Seems overkill to me as well.  We could try webargs for this sort of thing.